sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/BeamBuiltinAggregations.java

```diff
@@ -426,11 +515,55 @@ public Long mergeAccumulators(Iterable<Long> accums) {
     }
 
     @Override
-    public Long extractOutput(Long accum) {
+    public Long extractOutput(Long accumulator) {
       if (this.isEmpty) {
         return null;
       }
-      return accum;
+      return accumulator;
+    }
+  }
+
+  static CombineFn createLogicalAnd(Schema.FieldType fieldType) {
+    if (fieldType.getTypeName() == TypeName.BOOLEAN) {
+      return new LogicalAnd();
+    }
+    throw new UnsupportedOperationException(
+        String.format("[%s] is not supported in LOGICAL_AND", fieldType));
+  }
+
+  public static class LogicalAnd extends CombineFn<Boolean, Boolean, Boolean> {
+    private boolean isEmpty = true;
+
+    @Override
+    public Boolean createAccumulator() {
+      return Boolean.FALSE;
+    }
+
+    @Override
+    public Boolean addInput(Boolean mutableAccumulator, Boolean input) {
+      if (input != null) {
+        this.isEmpty = false;
+        return mutableAccumulator && input;
+      } else {
+        return null;
```

Comment:
The documentation for LOGICAL_AND says:

"Returns the logical AND of all non-NULL expressions. Returns NULL if there are zero input rows or expression evaluates to NULL for all rows."

In your current implementation, LOGICAL_AND would return NULL if *any* row was NULL. Please add some test with nulls to verify the correct behavior:

1. returns null if there are no input rows
2. returns null if all input rows are null
3. returns logical AND of all non-NULL expressions if at least one input row is null and at least one input row is non-null

https://github.com/google/zetasql/blob/master/docs/functions-and-operators.md#logical_and